python/ray/data/read_api.py

```diff
@@ -118,12 +119,17 @@ def from_items(
         items: List of local Python objects.
         parallelism: The amount of parallelism to use for the dataset.
             Parallelism may be limited by the number of items.
-        output_arrow_format: If True, always return data in Arrow format, raising an
-            error if this is not possible. Defaults to False.
 
     Returns:
         MaterializedDataset holding the items.
     """
+    if output_arrow_format:
+        warnings.warn(
+            DeprecationWarning,
+            "`output_arrow_format` is deprecated. In Ray 2.5+, `from_items` always "
+            "tries to store items in Arrow format.",
```

Comment:
```suggestion
            "stores items in Arrow format.",
```

Reply:
Hmm this is only true in strict mode right? Technically, it is still supported if we disable strict mode.

Maybe we should force strict mode always in 2.6?

Reply:
@ericl I thought that was the plan?

Reply:
IIRC we say in our documentation that after 2.5 you can't disable strict mode.

Reply:
@amogkam I say "try" because some types can't be represented in Arrow tables, and in that case `DelegatingBlockBuilder` stores the items in pandas DataFrames

Reply:
Sounds good,

1. Let's clarify in the docstring that the arg is only valid when strict mode is disabled.
2. Print the deprecation warning when the arg is set and strict mode is enabled.